core/modules/dashboard/dashboard.install

```diff
@@ -12,10 +12,40 @@ function dashboard_install() {
 }
 
 /**
- * Implements hook_uninstall().
+ * Implements hook_enable().
+ */
+function dashboard_enable() {
+  // Enable the Dashboard menu item (if it exists and is disabled).
+  $link = db_query("SELECT * FROM {menu_links} WHERE router_path LIKE 'admin/dashboard' AND module = 'system' AND hidden = 1")->fetchAssoc();
```

Comment:
It seems you're only using the mlid here, so probably best to just return that field rather than all of them...